docker-java-transport-zerodep/src/main/java/com/github/dockerjava/httpclient5/ZerodepDockerHttpClient.java

```diff
@@ -45,14 +49,26 @@ public Builder responseTimeout(Duration responseTimeout) {
             return this;
         }
 
+        public Builder connectionKeepAlive(Duration connectionKeepAlive) {
+            this.connectionKeepAlive = connectionKeepAlive;
+            return this;
+        }
+
+
+        public Builder validateAfterInactivity(Duration validateAfterInactivity) {
+            this.validateAfterInactivity = validateAfterInactivity;
```

Comment:
This looks sensible to me. I wondered how the Http Client actually tells if the connection is still valid. From the code I see that it tries a read with a short timeout, with the hope of inducing an IOException from a broken socket.

org.apache.hc.core5.http.impl.io.BHttpConnectionBase#isStale

Reply:
I had verified they worked in the bad case, but on further testing today it appears to work terribly in the case where the connection isn't closed. So testing against `docker` it blocks forever in `isStale` and with `podman` it blocks until the connection is closed by the server. So I think that solution is no good. I missed that the default stale check was disabled for this reason https://github.com/docker-java/docker-java/issues/1726, so I think that's eliminated as an option.